source/store_test.go

```diff
@@ -17,86 +17,73 @@ limitations under the License.
+	args := m.Called()
+	if args.Error(1) == nil {
+		m.client = args.Get(0).(kubernetes.Interface)
+		return m.client, nil
 	}
+	return nil, args.Error(1)
 }
 
-// testLookupMultiple tests that Sources can be looked up by providing multiple names.
-func testLookupMultiple(t *testing.T) {
-	for _, tc := range []struct {
-		title       string
-		registered  map[string]Source
-		names       []string
-		expectError bool
-	}{
-		{
-			"multiple registered sources are found by names",
-			map[string]Source{
-				"foo": &testutils.MockSource{},
-				"bar": &testutils.MockSource{},
-			},
-			[]string{"foo", "bar"},
-			false,
-		},
-		{
-			"multiple registered sources, one source not registered",
-			map[string]Source{
-				"foo": &testutils.MockSource{},
-				"bar": &testutils.MockSource{},
-			},
-			[]string{"foo", "bar", "baz"},
-			true,
-		},
-	} {
-		t.Run(tc.title, func(t *testing.T) {
-			for k, v := range tc.registered {
-				Register(k, v)
-			}
-
-			lookup, err := LookupMultiple(tc.names)
-			if tc.expectError {
-				require.Error(t, err)
-			} else {
-				require.NoError(t, err)
-				require.Len(t, lookup, len(tc.registered))
-				for _, source := range tc.registered {
-					assert.Contains(t, lookup, source)
-				}
-			}
-		})
-	}
+type ByNamesTestSuite struct {
+	suite.Suite
+}
+
+func (suite *ByNamesTestSuite) TestAllInitialized() {
+	mockClientProvider := new(MockClientProvider)
+	mockClientProvider.On("KubeClient").Return(fake.NewSimpleClientset(), nil)
+
+	sources, err := ByNames(mockClientProvider, []string{"service", "ingress", "fake"}, &Config{})
+	suite.NoError(err, "should not generate errors")
+	suite.Len(sources, 3, "should generate all three sources")
+}
+
+func (suite *ByNamesTestSuite) TestOnlyFake() {
+	mockClientProvider := new(MockClientProvider)
+	mockClientProvider.On("KubeClient").Return(fake.NewSimpleClientset(), nil)
+
+	sources, err := ByNames(mockClientProvider, []string{"fake"}, &Config{})
+	suite.NoError(err, "should not generate errors")
+	suite.Len(sources, 1, "should generate all three sources")
+	suite.Nil(mockClientProvider.client, "client should not be created")
+}
+
+func (suite *ByNamesTestSuite) TestSourceNotFound() {
+	mockClientProvider := new(MockClientProvider)
+	mockClientProvider.On("KubeClient").Return(fake.NewSimpleClientset(), nil)
+
+	sources, err := ByNames(mockClientProvider, []string{"foo"}, &Config{})
+	suite.Equal(err, ErrSourceNotFound, "should return sourcen not found")
+	suite.Len(sources, 0, "should not returns any source")
+}
+
+func (suite *ByNamesTestSuite) TestKubeClientFails() {
+	mockClientProvider := new(MockClientProvider)
+	mockClientProvider.On("KubeClient").Return(nil, errors.New("foo"))
+
+	_, err := ByNames(mockClientProvider, []string{"service"}, &Config{})
+	suite.NotNil(err, "should return an error if client cannot be created")
+
+	_, err = ByNames(mockClientProvider, []string{"ingress"}, &Config{})
```

Comment:
purely for coverage increase :P it tests the case when ingress client generation has an error